atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/CassandraMultinodeTodoEteTest.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2016 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.ete;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThan;
+import static org.hamcrest.core.IsCollectionContaining.hasItem;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.atlasdb.ete.EteSetup;
+import com.palantir.atlasdb.todo.ImmutableTodo;
+import com.palantir.atlasdb.todo.Todo;
+import com.palantir.atlasdb.todo.TodoResource;
+
+public class CassandraMultinodeTodoEteTest {
+    private static final List<String> CLIENTS = ImmutableList.of("ete1");
+
+    @ClassRule
+    public static final RuleChain COMPOSITION_SETUP = EteSetup.setupComposition(
+            "cassandra-multinode",
+            "docker-compose.multiple-cassandra.yml",
+            CLIENTS);
+
+    private static final List<String> CASSANDRA_NODES = ImmutableList.of("cassandra1", "cassandra2", "cassandra3");
+
+    private static final long MAX_CASSANDRA_NODE_DOWN_MILLIS = 30000;
+    private static final long MAX_CASSANDRA_NODES_RUNNING_MILLIS = 3000;
+
+    @Test
+    public void shouldRunTransactionsWithAllCassandraNodesRunningWithoutUnacceptableDelay()
+            throws InterruptedException {
+        TodoResource clientToSingleNode = EteSetup.createClientToSingleNode(TodoResource.class);
+
+        long transactionStartTime = System.currentTimeMillis();
+        assertAddTodoTransactionWasSuccessful(clientToSingleNode);
+        long transactionEndTime = System.currentTimeMillis();
+
+        long transactionTimeWithAllNodesRunning = transactionEndTime - transactionStartTime;
+
+        assertThat("transactionTimeWithAllNodesRunning",
+                transactionTimeWithAllNodesRunning,
+                is(lessThan(MAX_CASSANDRA_NODES_RUNNING_MILLIS)));
+    }
+
+    @Test
+    public void shouldRunTransactionsAfterCassandraNodeIsShutDownWithoutUnacceptableDelay()
+            throws InterruptedException {
+        TodoResource clientToSingleNode = EteSetup.createClientToSingleNode(TodoResource.class);
+
+        assertAddTodoTransactionWasSuccessful(clientToSingleNode);
+
+        String cassandraNodeToKill = getRandomCassandraNodeToShutdown();
```

Comment:
How did we pick these timings? We should be able to do a quorum write with 2 or 3 nodes available in roughly the same time range assuming we talk to a live node that is not partitioned away from the cluster. If we try talking to a dead or partitioned node, I could see needing to wait for timeout & retry. 

With the random based Russian roulette here, we get different behavior each time ETE runs -- a 1 in 3 chance the client we talk to is the one killed. We might want to expand the tests to cover these scenarios more explicitly.